### PR TITLE
EIP 158 + 160 protocol update

### PR DESCRIPTION
**Pull request by chriseth**
This implements https://github.com/ethereum/eips/issues/158 and https://github.com/ethereum/eips/issues/160 and https://github.com/ethereum/eips/issues/170

Still to do:
- [x] https://github.com/ethereum/eips/issues/155
- [x] Set transition block number in genesisInfo and tests

### CODE REVIEW

File: libethereum/Block.cpp [hunk 1]
```diff
@@ -688,7 +689,7 @@ void Block::performIrregularModifications()
 		Addresses allDAOs = childDaos();
 		for (Address const& dao: allDAOs)
 			m_state.transferBalance(dao, recipient, m_state.balance(dao));
-		m_state.commit();
+		m_state.commit(false);
```
Comment (chfast) [inline libethereum/Block.cpp]: These bool flags are very annoying. Can you try using `enum: bool`?

Comment (chriseth) [inline libethereum/Block.cpp]: Yes, let's convert this into an enum - although I'm not sure if `: bool` is required.

**Comment by chriseth** [comment]
Short test against solidity caused this:
```
eth: /home/christian/ethereum/cpp-ethereum/libethereum/Account.h:160: h256 dev::eth::Account::codeHash() const: Assertion `!isFreshCode()' failed.
```

**Comment by chfast** [comment]
What test?

**Comment by chriseth** [comment]
just the solidity end to end tests

**Comment by chriseth** [comment]
Sorry, false alarm, I think I did not include your fixes to the other PR.

**Comment by winsvega** [comment]
suicide issue test failing

**Comment by chriseth** [comment]
@chfast Build error on both windows and linux - could you please take a look?
```
In file included from /home/travis/build/ethereum/cpp-ethereum/libevm/VMFactory.cpp:23:0:

/home/travis/build/ethereum/cpp-ethereum/libevm/JitVM.h:3:20: fatal error: evmjit.h: No such file or directory

 #include <evmjit.h>
```

**Comment by chriseth** [comment]
Rebased and squashed some commits.

File: scripts/tests.bat [hunk 2]
```diff
@@ -33,6 +33,7 @@ if "%TESTS%"=="On" (
     REM Clone the end-to-end test repo, and point environment variable at it.
     cd ..
     git clone --depth 1 https://github.com/ethereum/tests.git
+    git checkout eip158
```
Comment (chriseth) [inline scripts/tests.bat]: Please fix the tests repo and remove this before merging.

Comment (winsvega) [inline scripts/tests.bat]: yes. this is for green tests

File: .gitmodules [hunk 3]
```diff
@@ -6,3 +6,7 @@
 	path = evmjit
 	url = https://github.com/ethereum/evmjit
 	branch = develop
+[submodule "libjson-rpc-cpp"]
```
Comment (chfast) [inline .gitmodules]: This is not needed in develop branch so it should not be here.

**Comment by axic** [comment]
@winsvega @chriseth commit 9d6a2934 rolled back the LLL changes :(

**Comment by chriseth** [comment]
Argh, could you please fix that, @winsvega ?

**Pull request merged**